### PR TITLE
Dockerize app 

### PR DESCRIPTION
**Pull request by gonzalesraul**
Dockerfile and Makefile added to the project to build a minimal docker container. (~22MB)

## Changelog :memo:

1. Added `Dockerfile` with the steps to build the docker image;
2. Added `Makefile` to facilitate run the docker steps;
3. Added `nginx.conf` to setup nginx process to run on the container;

## Steps to validate:

1. Run `make build` (_Alternatively_ `docker build -t magnobiet/marvel-app:1.0.0 -f Dockerfile .`)
2. Run `docker run -it --rm --name marvel -p 8080:80 magnobiet/marvel-app:1.0.0`
3. Open the navigator on http://localhost:8080 and check the results

Closes: #8 Dockerize App

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+COPY package.json ./
+RUN npm config set progress=false && npm config set depth 0 \
+    && npm install \
+    && mkdir /opt/app \
+    && cp -R ./node_modules /opt/app
+
+FROM dependencies as builder
+
+WORKDIR /opt/app
+COPY . .
```
Comment (fabiojb) [inline Dockerfile]: Faz sentido esse copy com `src` e o `dest` iguais?

**Comment by magnobiet** [comment]
Great job @gonzalesraul, thank you so much.

**Comment by magnobiet** [comment]
I see two little problems with `nginx`, you can help me?

The problems are related to routes and API proxy.

## Steps to reproduce:

1. Build and run image
2. Navigate to `http://localhost:8080/`
3. Click on **Characters** menu
4. Refresh page returns `404 Not Found`

Note that Vue Router is using `mode: 'history'`.

A second problem is related to `/api` route.

On **Characters** page, a request to `http://localhost:8080/api/characters?apikey=...` is made, this is a proxy pass to `https://gateway.marvel.com/v1/public/`. At `nginx` something like that:

```nginx
location /api {
    proxy_pass https://gateway.marvel.com/v1/public/;
}
```

I plan to create another issue to solve that. What do you think?

Can you help me to resolve that?

**Comment by gonzalesraul** [comment]
Sure @magnobiet, I can help you with it, just one thing about the proxy pass. Does that destination allow CORS?

**Comment by magnobiet** [comment]
Yep, CORS is allowed, but the initial idea is access via a proxy pass to `/api` route.

On local environment, the proxy is made on `webpack` [config/dev.env.js#L8](https://github.com/magnobiet/marvel-app/blob/master/config/dev.env.js#L8)

At Heroku I used using `buildpack-static` ([static.json#L9](https://github.com/magnobiet/marvel-app/blob/master/static.json#L9))

Example running at https://marvel-app-vue.herokuapp.com/.

I do approve of this issue and I create another to resolve this. What do you think?

**Comment by gonzalesraul** [comment]
Hi @magnobiet, I've already make an amend on this commit with the fix. 

As the feature had been broken (character view), I think the issue was not resolved in first place, for that reason, I would suggest to keep all the changes related on the first issue and PR. 

**What I did?**
I had to use env vars to resolve the issue:
`API_ENDPOINT` = `https://gateway.marvel.com/v1/public/`.

This can be overwritten at anytime during `docker run`, for instance:
```sh
$ docker run -it --rm --name marvel -p 8080:80 -e API_ENDPOINT='http://localhost:9000/mock/marvel/v1/' magnobiet/marvel-app:1.0.0
```
You can read more about env vars and nginx runing on containers on the link below:
https://docs.docker.com/samples/library/nginx/#using-environment-variables-in-nginx-configuration

Please review and let me know.

**Comment by magnobiet** [comment]
@gonzalesraul has a problem with `nginx` proxy pass. All routes except `/api/` redirect to `localhost`.

Examples:
- On access `http://localhost:8080/:anything`, redirect to `http://localhost/`;
- On access `http://localhost:8080/api/:anything` is working.

**Pull request closed without merging**